uniffi_bindgen/src/bindings/python/gen_python.rs

```diff
@@ -87,94 +87,82 @@ mod filters {
-                "lowerOptional({}, lambda buf, v: {})",
-                nm,
-                lower_into_py(&"buf", &"v", type_)?
+            Type::Int8
+            | Type::UInt8
+            | Type::Int16
+            | Type::UInt16
+            | Type::Int32
+            | Type::UInt32
+            | Type::Int64
+            | Type::UInt64
+            | Type::Float32
+            | Type::Float64 => nm.to_string(),
+            Type::Boolean => format!("(1 if {} else 0)", nm),
+            Type::String => format!("RustString.allocFromString({})", nm),
+            Type::Enum(_) => format!("({}.value)", nm),
+            Type::Object(_) => format!("({}._uniffi_handle)", nm),
+            Type::Error(_) => panic!("No support for lowering errors, yet"),
+            Type::Record(_) | Type::Optional(_) | Type::Sequence(_) => format!(
+                "RustBuffer.allocFrom{}({})",
+                class_name_py(&type_.unique_name())?,
+                nm
             ),
-            _ => panic!("[TODO: lower_py({:?})]", type_),
-        })
-    }
-
-    pub fn lowers_into_size_py(
-        nm: &dyn fmt::Display,
-        type_: &Type,
-    ) -> Result<String, askama::Error> {
-        let nm = var_name_py(nm)?;
-        Ok(match type_ {
-            Type::UInt32 => "4".to_string(),
-            Type::Float64 => "8".to_string(),
-            Type::String => format!("4 + len({}.encode('utf-8'))", nm),
-            Type::Record(type_name) => format!("{}._lowersIntoSize({})", type_name, nm),
-            _ => panic!("[TODO: lowers_into_size_py({:?})]", type_),
-        })
-    }
-    pub fn lower_into_py(
-        nm: &dyn fmt::Display,
-        target: &dyn fmt::Display,
-        type_: &Type,
-    ) -> Result<String, askama::Error> {
-        let nm = var_name_py(nm)?;
-        Ok(match type_ {
-            Type::Float64 => format!("{}.putDouble({})", target, nm),
-            Type::UInt32 => format!("{}.putInt({})", target, nm),
-            Type::String => format!("{}.putString({})", target, nm),
-            Type::Record(type_name) => format!("{}._lowerInto({}, {})", type_name, nm, target),
-            _ => panic!("[TODO: lower_into_py({:?})]", type_),
         })
     }
 
     pub fn lift_py(nm: &dyn fmt::Display, type_: &Type) -> Result<String, askama::Error> {
         Ok(match type_ {
-            Type::UInt32 | Type::UInt64 | Type::Float32 | Type::Float64 | Type::Boolean => {
-                format!("{}", nm)
-            }
-            Type::Enum(type_name) => format!("{}({})", type_name, nm),
-            Type::String => format!("liftString({})", nm),
-            Type::Record(type_name) => format!("{}._lift({})", type_name, nm),
-            Type::Optional(type_) => format!(
-                "liftOptional({}, lambda buf: {})",
-                nm,
-                lift_from_py(&"buf", type_)?
-            ),
-            Type::Sequence(type_) => format!(
-                "liftSequence({}, lambda buf: {})",
```

Comment:
Python lambdas are a pain, much of this is motivated by not wanting to build an elaborate series of nested lambdas for sequences-of-optional-types and things like that.